pennylane/ops/arithmetic/adjoint.py

```diff
@@ -0,0 +1,433 @@
+        r"""Multiplier and shift for the given parameter, based on its gradient recipe.
+
+        Args:
+            idx (int): parameter index within the operation
+
+        Returns:
+            list[[float, float, float]]: list of multiplier, coefficient, shift for each term in the gradient recipe
+
+        Note that the default value for ``shift`` is None, which is replaced by the
+        default shift :math:`\pi/2`.
+        """
+        return self.base.get_parameter_shift(idx)
+
+    @property
+    def parameter_frequencies(self):
+        r"""Returns the frequencies for each operator parameter with respect
+        to an expectation value of the form
+        :math:`\langle \psi | U(\mathbf{p})^\dagger \hat{O} U(\mathbf{p})|\psi\rangle`.
+
+        These frequencies encode the behaviour of the operator :math:`U(\mathbf{p})`
+        on the value of the expectation value as the parameters are modified.
+        For more details, please see the :mod:`.pennylane.fourier` module.
+
+        Returns:
+            list[tuple[int or float]]: Tuple of frequencies for each parameter.
+            Note that only non-negative frequency values are returned.
+
+        **Example**
+
+        >>> op = qml.CRot(0.4, 0.1, 0.3, wires=[0, 1])
+        >>> op.parameter_frequencies
+        [(0.5, 1), (0.5, 1), (0.5, 1)]
+
+        For operators that define a generator, the parameter frequencies are directly
+        related to the eigenvalues of the generator:
+
+        >>> op = qml.ControlledPhaseShift(0.1, wires=[0, 1])
+        >>> op.parameter_frequencies
+        [(1,)]
+        >>> gen = qml.generator(op, format="observable")
+        >>> gen_eigvals = qml.eigvals(gen)
+        >>> qml.gradients.eigvals_to_frequencies(tuple(gen_eigvals))
+        (1.0,)
+
+        For more details on this relationship, see :func:`.eigvals_to_frequencies`.
+        """
+        return self.base.parameter_frequencies
+
+
+def adjoint(fn, lazy=True):
+    """Create a function that applies the adjoint of the provided operation or template.
+
+    This transform can be used to apply the adjoint of an arbitrary sequence of operations.
+
+    Args:
+        fn (function): A single operator or a quantum function that
+            applies quantum operations.
+
+    Keyword Args:
+        lazy=True (bool): If the transform is behaving lazily, all operations are wrapped in a `Adjoint` class
+            and handled later. If ``lazy=False``, operation-specific adjoint decompositions are first attempted.
+
+    Returns:
+        function: A new function that will apply the same operations but adjointed and in reverse order.
+
+    .. note::
+
+        While the adjoint and inverse are identical for Unitary gates, not all possible operators are Unitary.
+        This transform can also act on Channels and Hamiltonians, for which the inverse and adjoint are different.
+
+    .. seealso:: :class:`.ops.arithmetic.Adjoint` and :meth:`.operation.Operator.adjoint`
```

Comment:
The first link isn't working for me 🤔 Perhaps maybe just `.Adjoint` might work?

Reply:
Oh, this could be because `Adjoint` isn't showing in the docs yet?

Reply:
I think I forgot the `~`. I'll add those in.

But `Adjoint` isn't showing up in the docs yet.  How to document the `Adjoint` class is still an unresolved issue.